presto-hive/src/test/java/com/facebook/presto/hive/TestS3SelectRecordCursor.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.TypeManager;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.mapred.RecordReader;
+import org.joda.time.DateTimeZone;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static com.facebook.presto.hive.HiveType.HIVE_INT;
+import static com.facebook.presto.hive.HiveType.HIVE_STRING;
+import static com.facebook.presto.hive.S3SelectRecordCursor.updateSplitSchema;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static java.util.Arrays.asList;
+import static java.util.Collections.EMPTY_LIST;
+import static java.util.Collections.singletonList;
+import static org.apache.hadoop.hive.serde.serdeConstants.LIST_COLUMNS;
+import static org.apache.hadoop.hive.serde.serdeConstants.LIST_COLUMN_TYPES;
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_DDL;
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_LIB;
+import static org.mockito.Mockito.mock;
+import static org.testng.Assert.assertEquals;
+
+public class TestS3SelectRecordCursor
+{
+    private static final String APACHE_HADOOP_HIVE = "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe";
```

Comment:
`APACHE_HADOOP_HIVE` -> `LAZY_SERDE_CLASS_NAME` or something like that to make it more descriptive.

Reply:
Refactored